backend/routing/Router.go

```diff
@@ -0,0 +1,60 @@
+package routing
+
+import (
+	"log"
+	"net/http"
+
+	"github.com/fourtf/studyhub/controllers"
+	"github.com/fourtf/studyhub/middlewares"
+	"github.com/gorilla/mux"
+	"github.com/justinas/nosurf"
+)
+
+var router *mux.Router = mux.NewRouter().StrictSlash(true)
+
+//SetupRouter initializes the router by assigning paths and middlewares
+func SetupRouter() *mux.Router {
```

Comment:
Not sure if splitting this into functions is good here. Generally simple functions can be longer than complex functions and I would consider adding the routes to be a simple operation.

See also: https://www.kernel.org/doc/html/v4.10/process/coding-style.html#functions